packages/beacon-node/src/chain/factory/block/body.ts

```diff
@@ -122,39 +122,26 @@ export async function assembleBody<T extends BlockType>(
           feeRecipient
         );
       }
+
```

Comment:
@g11tech this call here is confusing
```

      // This path will not be used in the production, but is here just for merge mock
      // tests because merge-mock requires an fcU to be issued prior to fetch payload
      // header.
      if (chain.executionBuilder.issueLocalFcUForBlockProduction) {
        await prepareExecutionPayload(
          chain,
          safeBlockHash,
          finalizedBlockHash ?? ZERO_HASH_HEX,
          currentState as CachedBeaconStateBellatrix,
          feeRecipient
        );
      }
```
You say that this needs to can fcU, but instead calls the entire `prepareExecutionPayload` flow which does more things

Reply:
@dapplion this was just a hack to run merge mock, i wanted to it for our mev boost tests, previously `prepareExecutionPayload` was just issuing fcU (and may be catching the payloadId) but pretty harmless for a test.